inference-engine/src/transformations/src/transformations/common_optimizations/pull_transpose_through_fq.cpp

```diff
@@ -9,56 +9,42 @@
-    auto data5 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto fq = std::make_shared<ngraph::opset1::FakeQuantize>(data1, data2, data3, data4, data5, 1);
-    auto transpose_order = std::make_shared<pattern::op::Label>(element::i64, Shape{4});
-    auto transpose = std::make_shared<ngraph::opset1::Transpose>(fq, transpose_order);
+    auto m_fq = pattern::wrap_type<opset1::FakeQuantize>({pattern::any_input(pattern::has_static_rank()),
+                                                          pattern::any_input(pattern::has_static_rank()),
+                                                          pattern::any_input(pattern::has_static_rank()),
+                                                          pattern::any_input(pattern::has_static_rank()),
+                                                          pattern::any_input(pattern::has_static_rank())},
+                                                          pattern::consumers_count(1));
+    auto m_transpose = pattern::wrap_type<opset1::Transpose>({m_fq, pattern::wrap_type<opset1::Constant>()});
 
-    ngraph::graph_rewrite_callback callback = [](pattern::Matcher& m) {
-        auto transpose = ngraph::as_type_ptr<ngraph::opset1::Transpose>(m.get_match_root());
-        if (!transpose) {
-            return false;
-        }
-
-        auto const_node = transpose->input(1).get_source_output().get_node_shared_ptr();
-        auto const_order = ngraph::as_type_ptr<ngraph::opset1::Constant>(const_node);
-        if (!const_order) {
-            return false;
-        }
-
-        auto fq_node = transpose->input(0).get_source_output().get_node_shared_ptr();
-        auto fq = ngraph::as_type_ptr<ngraph::opset1::FakeQuantize>(fq_node);
-        if (!fq || fq->output(0).get_target_inputs().size() != 1) {
-            return false;
-        }
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto & pattern_map = m.get_pattern_value_map();
+        auto transpose = pattern_map[m_transpose].get_node_shared_ptr();
+        auto fq = pattern_map[m_fq].get_node_shared_ptr();
 
-        auto input_shape = fq->input(0).get_source_output().get_shape();
+        auto input_shape = fq->input(0).get_partial_shape().rank().get_length();
 
         ngraph::NodeVector new_ops;
         ngraph::OutputVector fq_inputs;
         for (size_t i = 0; i < fq->inputs().size(); ++i) {
-            std::shared_ptr<ngraph::Node> fq_input;
-            fq_input = fq->input(i).get_source_output().get_node_shared_ptr();
-            auto fq_input_shape = fq_input->get_shape();
+            auto fq_input = fq->input_value(i);
+            auto fq_input_rank = fq_input.get_partial_shape().rank().get_length();
             std::vector<int64_t> unsqueeze_axes;
-            for (size_t j = 0; j < input_shape.size() - fq_input_shape.size(); ++j) {
+            for (size_t j = 0; j < input_shape - fq_input_rank; ++j) {
```

Comment:
What if `input_shape` is less than `fq_input_rank`?

Reply:
rank().get_length() returns int64_t so the comparison with size_t will be ok in case of negative value

Reply:
Ok, I thought that rank has a type uint64_t